crates/proto/src/domain/note.rs

```diff
@@ -8,6 +10,18 @@ use crate::{
     generated::note as proto,
 };
 
+/// A note which has been committed to the chain and therefore has an associated index in a specific block and an inclusion proof.
+#[derive(Debug, Clone, PartialEq)]
+pub struct CommittedNote {
+    pub block_num: BlockNumber,
+    // TODO: can this be BlockNoteIndex? If so, how to convert to an absolute index.
+    pub note_index: usize,
+    pub note_id: Digest,
+    pub metadata: NoteMetadata,
+    pub details: Option<Vec<u8>>,
+    pub merkle_path: MerklePath,
+}
```

Comment:
Because I haven't implemented the actual network tx builder yet, I'm unsure if this is what is needed on that end. I also don't know how to get some of these values, e.g. the merkle path and note index.

Reply:
As mentioned in another comment, I think we may be able to get away with just sending notes (rather than including things like Merkle path, note index etc.).